src/wp-includes/class-wp-query.php

```diff
@@ -3166,7 +3166,8 @@ public function get_posts() {
 			"{$wpdb->posts}.ID",
 		);
 
-		if ( ! in_array( $fields, $cacheable_field_values, true ) ) {
+		$id_posts_are_cacheable = in_array( $fields, $cacheable_field_values, true );
+		if ( ! $id_posts_are_cacheable ) {
```

Comment:
Technically speaking, this is a potential issue currently: The posts really are only cacheable if _all_ post fields are queried. If the query queries only `ID`, or only `ID` and `post_parent`, we must not cache those results as it would be incomplete post objects.

Right now this wouldn't cause real problems since the `update_post_cache()` call below won't be reached in the other two cases anyway since for both of them (`ids` and `id=>parent`) early returns are present. But it's a risk so I'd prefer to set this variable only to true if really _all_ post fields are queried.

This is a completely separate check though, so I think we can revert this and apply the other check only where it's needed further below.
```suggestion
		if ( ! in_array( $fields, $cacheable_field_values, true ) ) {
```